library/core/src/alloc/layout.rs

```diff
@@ -514,4 +542,190 @@ mod verify {
             assert_eq!(layout.align(), a);
         }
     }
+
+    // pub const fn size(&self) -> usize
+    #[kani::proof]
+    pub fn check_size() {
+        let s = kani::any::<usize>();
+        let a = kani::any::<usize>();
+
+        unsafe {
+            let layout = Layout::from_size_align_unchecked(s, a);
+            assert_eq!(layout.size(), s);
+        }
+    }
+
+    // pub const fn align(&self) -> usize
+    #[kani::proof_for_contract(Layout::align)]
+    pub fn check_align() {
+        let s = kani::any::<usize>();
+        let a = kani::any::<usize>();
+
+        unsafe {
+            let layout = Layout::from_size_align_unchecked(s, a);
+            assert!(layout.align().is_power_of_two());
+        }
+    }
+
+    // pub const fn new<T>() -> Self
+    #[kani::proof]
+    pub fn check_new_i32() {
+        let layout = Layout::new::<i32>();
+        assert_eq!(layout.size(), 4);
+        assert!(layout.align().is_power_of_two());
+    }
+
+    // pub const fn for_value<T: ?Sized>(t: &T) -> Self
+    #[kani::proof_for_contract(Layout::for_value)]
+    pub fn check_for_value_i32() {
+        let array : [i32; 2] = [1, 2];
+        let layout = Layout::for_value::<[i32]>(&array[1 .. 1]);
+        assert!(layout.align().is_power_of_two());
+    }
+
+    // pub const unsafe fn for_value_raw<T: ?Sized>(t: *const T) -> Self
+    #[kani::proof_for_contract(Layout::for_value_raw)]
+    pub fn check_for_value_raw_i32() {
+        unsafe {
+            let layout = Layout::for_value_raw::<[i32]>(&[] as *const [i32]);
```

Comment:
For unsafe functions, it would be great if we could cover different safety scenarios.

Reply:
Hmm, yes, but how many would we consider "good enough?" And for this one specifically I believe we are lacking the means to express the different cases as `requires` clauses that the function's documentation calls out.

Reply:
I don't have a number for you, but I would consider in this case adding a few cases with types that have different layouts

Reply:
I have added another one, but need some help in producing objects for further two cases. TODOs added.